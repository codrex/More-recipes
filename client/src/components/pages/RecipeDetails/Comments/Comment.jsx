import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import Animation from '../../../common/Animation';
import getInitials from '../../../../utils/getInitials';

/**
 * Comment component
 * @param {object} props
 * @return {React} react element
 */
const Comment = props => (
  <Animation>
    <div className="media-object comment">
      <div className="avatar avatar-sm">{getInitials(props.fullname)}</div>
      <div className="media-body">
        <div className="d-flex justify-content-between align-items-center">
          <h4 className="no-margin">
            {`@${props.username}`}
          </h4>
          <span className="text-center bold time">
            {moment(props.createdAt).fromNow()}
          </span>
        </div>
        <p className="media-text no-margin">
          {props.review}
        </p>
      </div>
    </div>
  </Animation>
);

Comment.propTypes = {
  review: PropTypes.string.isRequired,
  fullname: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired
};

export default Comment;

