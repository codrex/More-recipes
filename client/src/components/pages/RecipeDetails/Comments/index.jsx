import React from 'react';
import PropTypes from 'prop-types';
import NotFound from '../../../common/NotFound';
import Comment from './Comment';
import Loader from '../../../common/Loader';

/**
 * Comments component
 * @param {object} props
 * @return {React} react element
 */
const Comments = (props) => {
  const { comments, isLoading } = props;
  return (
    <div className="col-12 " style={{ minHeight: '63vh' }}>
      {comments.map(({
        id,
        reviewer,
        review,
        createdAt
      }, i) => (
        <Comment
          key={id}
          review={review}
          createdAt={createdAt}
          fullname={reviewer.fullname}
          username={reviewer.username}
          index={i}
        />
      )
      )}
      {comments.length === 0 && !isLoading &&
        <NotFound message="no review found" />
      }
      { isLoading && <Loader loading={isLoading} />
      }
    </div>
  );
};

Comments.defaultProps = {
  comments: [],
  createdAt: '',
  isLoading: true
};
Comments.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.shape),
  createdAt: PropTypes.string,
  isLoading: PropTypes.bool,
};

export default Comments;
