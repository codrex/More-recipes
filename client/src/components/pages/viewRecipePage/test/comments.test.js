import React from 'react';
import renderer from 'react-test-renderer';
import Comments from '../comments/comments';

const props = {
  comments: [{
    id: 50,
    review: 'resioioioioi',
    ReviewerId: 1,
    RecipeId: 1,
    reviewer: {
      id: 1,
      username: 'testi',
      fullname: 'test tester ',
      profilePix: 'UNKNOWN',
      email: 'test.test@gmail.com',
    }
  }],
};

describe('Comments display component ', () => {
  test('expected to match empty snapshot', () => {
    const component = renderer.create(<Comments {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
});
