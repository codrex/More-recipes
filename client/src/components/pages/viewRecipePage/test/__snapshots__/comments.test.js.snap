// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments display component  expected to match empty snapshot 1`] = `
<div
  className="col-12"
>
  <div
    className="media-object comment"
  >
    <div
      className="avatar avatar-sm"
    >
      TT
    </div>
    <div
      className="media-body"
    >
      <h1>
        test tester 
      </h1>
      <h6
        className="text-dark lead"
      >
        @testi
      </h6>
      <p
        className="media-text no-margin"
      >
        resioioioioi
      </p>
    </div>
  </div>
</div>
`;
